src/core/pointcloud/qgspointclouddataprovider.h

```diff
@@ -157,7 +157,7 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
      *
      * \note Not available in Python bindings
      */
-    virtual QgsPointCloudIndex *index() const SIP_SKIP {return nullptr;}
+    virtual QgsPointCloudIndex index() const SIP_SKIP {return nullptr;}
```

Comment:
Let's make this a little clearer:
```suggestion
    virtual QgsPointCloudIndex index() const SIP_SKIP {return QgsPointCloudIndex( nullptr );}
```